src/main/java/io/usethesource/vallang/impl/primitive/StringValue.java

```diff
@@ -1380,6 +1438,72 @@ public void write(Writer w) throws IOException {
             assert indents.isEmpty();
         }
 
+        @Override
```

Comment:
I think this is correct, although I see we need to work hard with the surrogate pairs. Maybe If we would always read until `len - 1` unless the last character is a newline, and the int iterator always provides codepoints, there are no further special cases.

Reply:
But before we merge we should find an implemention that can say least stream line by line.  I suspect that the current experiment with the int iterator will slow down the stream to the point that it doesn't make sense to stream anymore.

Reply:
Well, we need to do work with surrogate pairs since every int the iterator returns can be either 1 or 2 chars. I initially had some code that would write 2 chars if there was room, but that just lead to duplication of logic with the `queedLowSurrogate` field. So I rewrote that to be a bit less wordy.